### PR TITLE
Feature/service status update to hmi

### PR DESCRIPTION
**Pull request by mked-luxoft**
Provides the information for  #[2791](https://github.com/smartdevicelink/sdl_core/issues/2791)

Proposal 
This PR is **ready** for review.

### Summary
Create OnServiceUpdate folder; update config json with OnServiceUpdate

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by mked-luxoft** [comment]
@yang1070 This PR was updated to comply with proposal updates. Please review.

**Comment by AStasiuk** [comment]
@theresalech, this PR is already reviewed and approved by Ford (by @yang1070 ) and ready for Livio review.

File: docs/Common/Enums/index.md [hunk 1]
```diff
@@ -1203,4 +1203,31 @@
...
+|VIDEO|0|Refers to the Video service.|
+|AUDIO|1|Refers to the Audio service.|
+|RPC|2|Refers to RPC service.|
+
+### ServiceEvent
+
+|Name|Value|Description|
+|:---|:----|:----------|
+|REQUEST_RECEIVED|0|When a request for a Service is received.|
+|REQUEST_ACCEPTED|1|When a request for a Service is Accepted.|
```
Comment (jacobkeeler) [inline docs/Common/Enums/index.md]:
```suggestion
|REQUEST_ACCEPTED|1|When a request for a Service is accepted.|
```

File: docs/Common/Enums/index.md [hunk 2]
```diff
@@ -1203,4 +1203,31 @@
...
+|REQUEST_REJECTED|2|When a request for a Service is Rejected.|
```
Comment (jacobkeeler) [inline docs/Common/Enums/index.md]:
```suggestion
|REQUEST_REJECTED|2|When a request for a Service is rejected.|
```

File: docs/BasicCommunication/OnServiceUpdate/index.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+Display a pop-up providing accurate feedback to the user on the status of the system or what steps to take in case of an error.
+
+!!!
+
+### Notification
+When a mobile navigation application is activated and sends a request for the Video Service there is a series of steps taken. 
+These steps include:
+
+* getting the current [system time](../GetSystemTime/index.md)
```
Comment (jacobkeeler) [inline docs/BasicCommunication/OnServiceUpdate/index.md]:
```suggestion
* getting the current [system time](../getsystemtime) 
```

File: docs/BasicCommunication/OnServiceUpdate/index.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+Processing these steps SDL sends OnServiceUpdate notification to HMI to provide accurate feedback on the status of the system.  
+
+#### Parameters
+
+|Name|Type|Mandatory|Additional|
+|:---|:---|:--------|:---------|
+|serviceType|[Common.ServiceType](../../common/enums/#servicetype)|true||
+|serviceEvent|[Common.ServiceEvent](../../common/enums/#serviceevent)|false||
+|reason|[Common.ServiceStatusUpdateReason](../../enums/#servicestatusupdatereason)|false||
```
Comment (jacobkeeler) [inline docs/BasicCommunication/OnServiceUpdate/index.md]:
```suggestion
|reason|[Common.ServiceStatusUpdateReason](../../common/enums/#servicestatusupdatereason)|false||
```

File: docs/BasicCommunication/OnServiceUpdate/index.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+|appID|Integer|false||
+
+_Note:_ the only time when SDL would not be able provide the `appID` would be during the first StartService request for the RPC service before RAI was sent.
```
Comment (jacobkeeler) [inline docs/BasicCommunication/OnServiceUpdate/index.md]:
```suggestion
!!! note
The only time when SDL would not be able provide the `appID` would be during the first StartService request for the RPC service before RAI was sent.
!!!
```

File: docs/BasicCommunication/OnServiceUpdate/index.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+#### JSON Example Notification
+
+```json
+{
+    "jsonrpc": "2.0",
+    "method": "BC. OnServiceUpdate",
+    "params": {
+        "serviceType": "RPC",
+        "serviceEvent":"REQUEST_RECEIVED"
+      }
```
Comment (jacobkeeler) [inline docs/BasicCommunication/OnServiceUpdate/index.md]:
```suggestion
    }
```

File: docs/BasicCommunication/OnServiceUpdate/index.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+OnServiceUpdate
+![OnServiceUpdate](assets/OnServiceUpdate.png)
+|||
+
+#### JSON Example Notification
+
+```json
+{
+    "jsonrpc": "2.0",
+    "method": "BC. OnServiceUpdate",
```
[the remark this answers is not in the document]
Comment (KhrystynaDubovyk) [reply docs/BasicCommunication/OnServiceUpdate/index.md]: @jacobkeeler, thanks, please see fixed in https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/154/commits/9eb2e45b717afae0ac4a2cc6dbca2b120f41ac7f

**Pull request merged**